src/components/delegate/delegate.component.js

```diff
@@ -13,8 +13,12 @@ const DelegateConstructor = function ($rootScope, $stateParams, $location, $http
         }).then(resp => {
             if (resp.data.success) {
                 vm.address = resp.data;
+
+                if (!vm.address.delegate) {
+                    $state.go('address', { address: $stateParams.delegateId });
+                }
             } else {
-                throw 'Account was not found!';
+                $state.go('home');
```

Comment:
The reason I didn't redirect to 404 is that there are some problems with 404 component I prefer to solve them in a dedicated ticket.